src/dftbp/api/mm/dftbplus.h

```diff
@@ -386,6 +406,48 @@ void dftbp_get_gross_charges(DftbPlus *instance, double *charges);
 void dftbp_get_cm5_charges(DftbPlus *instance, double *charges);
 
 
+/**
+ * Queries the reference valence charges of the atoms (effective Z).
+ *
+ * \param[inout] instance Handler of the DFTB+ instance.
+ *
+ * \param[out] charges reference charges on each atom. Shape: [natom].
+ */
+void dftbp_get_ref_charges(DftbPlus *instance, double *charges);
+
+
+/**
+ * Sets the reference valence charges of the atoms (effective Z).
+ *
+ * \param[inout] instance Handler of the DFTB+ instance.
+ *
+ * \param[in] charges reference charges on each atomic shell. Shape: [natom].
+ */
+void dftbp_set_ref_charges(DftbPlus *instance, double *charges);
+
+
+/**
+ * Sets the net charge of the QM system.
+ *
+ * \param[inout] instance Handler of the DFTB+ instance.
+ *
+ * \param[in] charge of the system (+ve value being electron deficient)
+ */
+void dftbp_set_net_charge(DftbPlus *instance, double *charge);
```

Comment:
What is the benefit in passing a pointer to the charge instead of the charge itself?